src/resolvers/customFieldResolvers.ts

```diff
@@ -422,15 +428,20 @@ export class KandelHomePageResolver {
         }
       }
     })
-
+    const rateMap = new Map<string, number>();
     return (await Promise.all(kandels.map(async kandel => {
+      rateMap.get(kandel.baseToken.id) ?? rateMap.set(kandel.baseToken.id, await fetchTokenPriceIn(kandel.baseToken, 'USDC'));
```

Comment:
is fetchTokenPriceIn an actual fetch or based on something in the model? I would have expected the UI to do the conversion.

Reply:
Hmm, okay. That complicates thinks. E.g. for return i also do this. If do not, then it means the UI should calculate the return aswell?